From a77debe1e3b3823e18d8ef021b694c42b1515dda Mon Sep 17 00:00:00 2001
From: Elvis Dowson <elvis.dowson@gmail.com>
Date: Wed, 22 Aug 2012 03:22:42 +0400
Subject: [PATCH 3/3] xilinxfb: Fix regression in call to XPS TFT controller
 write API.

This commit fixes a regression in xilinx_fb_out_be32(), where an incorrect
call to the XPS TFT controller's write API, caused the display not to work.

This fix has been adapted from linux-xilinx-3.0.0.

Signed-off-by: Elvis Dowson <elvis.dowson@gmail.com>
---
 drivers/video/xilinxfb.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/xilinxfb.c b/drivers/video/xilinxfb.c
index b800c9c..1808452 100644
--- a/drivers/video/xilinxfb.c
+++ b/drivers/video/xilinxfb.c
@@ -154,7 +154,7 @@ static void xilinx_fb_out_be32(struct xilinxfb_drvdata *drvdata, u32 offset,
 				u32 val)
 {
 	if (drvdata->flags & PLB_ACCESS_FLAG)
-		iowrite32(val, drvdata->regs + (offset << 2));
+		out_be32(drvdata->regs + (offset << 2), val);
 #ifdef CONFIG_PPC_DCR
 	else
 		dcr_write(drvdata->dcr_host, offset, val);
-- 
1.7.9.5

